cfme/services/catalogs/catalog.py

```diff
@@ -1,115 +1,104 @@
-# -*- coding: utf-8 -*-
-from functools import partial
-
+from widgetastic.widget import Text
+from widgetastic_patternfly import Button, Input
 from navmazing import NavigateToAttribute, NavigateToSibling
-
-from cfme import web_ui
-from cfme.exceptions import DestinationNotFound
-from cfme.fixtures import pytest_selenium as sel
-from cfme.web_ui import accordion, flash, form_buttons, Input
-from cfme.web_ui import toolbar as tb, paginator, CheckboxTable, match_location
-from utils.appliance import Navigatable
-from utils.appliance.implementations.ui import navigator, navigate_to, CFMENavigateStep
-from utils.blockers import BZ
-from utils.pretty import Pretty
 from utils.update import Updateable
+from utils.pretty import Pretty
+from utils.appliance import Navigatable
+from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
 from utils import version
 
 
-cfg_btn = partial(tb.select, "Configuration")
-catalog_tree = partial(accordion.tree, "Catalogs")
+from . import ServicesCatalogView
 
-listview_table = CheckboxTable(table_locator='//div[@id="list_grid"]/table')
 
-item_multiselect = web_ui.MultiSelect(
-    "//select[@id='available_fields']",
-    "//select[@id='selected_fields']",
-    "//div[@id='column_lists']//a[contains(@href, 'button=right')]/img",
-    "//div[@id='column_lists']//a[contains(@href, 'button=left')]/img")
+class CatalogForm(ServicesCatalogView):
+    title = Text('#explorer_title_text')
 
-form = web_ui.Form(
-    fields=[('name_text', Input("name")),
-            ('description_text', Input("description")),
-            ('button_multiselect', item_multiselect),
-            ('add_button', form_buttons.add),
-            ('save_button', form_buttons.save),
-            ('cancel_button', form_buttons.cancel)])
+    name_text = Input(name='name')
+    description_text = Input(name="description")
 
-item_form = web_ui.Form(
-    fields=[('type_select', "//select[@id='st_prov_type']"),
-            ('name_text', Input("name")),
-            ('description_text', Input("description")),
-            ('display_checkbox', Input("display")),
-            ('add_button', form_buttons.add)])
+    add_button = Button('Add')
+    save_button = Button('Save')
+    cancel_button = Button('Cancel')
 
-match_page = partial(match_location, controller='catalog', title='Catalogs')
 
+class CatalogsView(ServicesCatalogView):
+    title = Text("#explorer_title_text")
 
-def _all_catalogs_add_new(_):
-    if BZ(1213863).blocks:
-        sel.pytest.skip("Blocker on the bug 1213863")
-    catalog_tree('All Catalogs')
-    cfg_btn('Add a New Catalog')
+    @property
+    def is_displayed(self):
+        return (
+            super(CatalogsView, self).is_displayed and
```

Comment:
**REQUIRED**: Do not use super() like this. ServicesCatalogView should check that it is in ServicesCatalog and it should also check that accordion is enabled and such things - that it actually is usable for the steps to happen since that ensures correct matching of locations. I strongly advocate for creating a property, let's say `in_servies_catalogs` and use that one instead in the subclasses.

Then the parent class would look like.

```python
@property
def in_services_catalogs(self):
    return 'logged in as current user and correct menu item selected'

@property
def is_displayed(self):
    return self.in_services_catalogs and not accordion.is_dimmed and 'whatever else to ensure that the view is correct'
```

Then in the child class, like this one, you will just replace this line with

```python
self.in_services_catalogs and  # ...
```

You might see the super() usage in some places, yes, somewhere it makes sense and in the other places I am going to remove them :)